py/bitbox02/bitbox02/util.py

```diff
@@ -141,3 +144,68 @@ def set_app_static_privkey(self, privkey: bytes) -> None:
         data = self._read_cache()
         data.app_static_privkey = privkey
         self._write_cache(data)
+
+
+class BitBoxAppNoiseConfig(bitbox_api_protocol.BitBoxNoiseConfig):
+    """
+    Noise config that reads and stores the noise keys in the same location as the BitBoxApp.
+    """
+
+    def __init__(self) -> None:
+        system = platform.system()
+        if system == "Linux":
+            folder = Path(
+                os.environ.get(
+                    "XDG_CONFIG_HOME",
+                    # fallback
+                    os.path.join(os.environ["HOME"], ".config"),
+                )
+            )
+        elif system == "Darwin":
+            folder = Path(os.environ["HOME"]) / "Library" / "Application Support"
+        elif system == "Windows":
+            folder = Path(os.environ["APPDATA"])
+        else:
+            raise NotImplementedError("Unknown system: {}".format(system))
+        self._filename = folder / "bitbox" / "bitbox02" / "bitbox02.json"
+
+    def _read_cache(self) -> Any:
+        try:
+            with self._filename.open("r") as fileh:
+                return json.load(fileh)
+        except FileNotFoundError:
+            return {}
+        except json.JSONDecodeError:
+            return {}
+
+    def _write_cache(self, data: Dict[str, object]) -> None:
+        self._filename.parent.mkdir(parents=True, exist_ok=True)
+        with self._filename.open("w") as fileh:
+            json.dump(data, fileh)
+
+    def contains_device_static_pubkey(self, pubkey: bytes) -> bool:
+        data = self._read_cache()
+        return base64.b64encode(pubkey).decode() in data.get("deviceNoiseStaticPubkeys", [])
+
+    def add_device_static_pubkey(self, pubkey: bytes) -> None:
+        if not self.contains_device_static_pubkey(pubkey):
+            data = self._read_cache()
+            data.setdefault("deviceNoiseStaticPubkeys", [])
+            data["deviceNoiseStaticPubkeys"].append(base64.b64encode(pubkey).decode())
+            self._write_cache(data)
+
+    def get_app_static_privkey(self) -> Optional[bytes]:
+        data = self._read_cache()
+        if "appNoiseStaticKeypair" not in data:
+            return None
+
+        return base64.b64decode(data["appNoiseStaticKeypair"]["private"])
+
+    def set_app_static_privkey(self, privkey: bytes) -> None:
+        data = self._read_cache()
+        pubkey = KeyPair25519.from_private_bytes(privkey).public_bytes
+        data["appNoiseStaticKeypair"] = {
```

Comment:
Make a class attributes for `"appNoiseStaticKeypair"` and `"deviceNoiseStaticPubkeys"`? Would make it easier to proof-read and might avoid potential typos in the future.

Reply:
What do you mean? In the end it has to be in the dict serialized as json anyway?

Reply:
It's just they are repeating. I imagined something like this:

```python
class BitBoxAppNoiseConfig(bitbox_api_protocol.BitBoxNoiseConfig):
    DEVICE_NOISE_STATIC_PUBKEYS = "deviceNoiseStaticPubkeys"
    APP_NOISE_STATIC_KEYPAIR = "appNoiseStaticKeypair"
    ...
    data.get(DEVICE_NOISE_STATIC_PUBKEYS)
    data[APP_NOISE_STATIC_KEYPAIR] = ...
```

Reply:
Well, I meant with `self.XXX` :)

Reply:
ah of course. done